cpp/log/cert.cc

```diff
@@ -92,6 +112,26 @@ Cert::Status Cert::LoadFromDerString(const std::string& der_string) {
   return Cert::TRUE;
 }
 
+Cert::Status Cert::LoadFromDerBio(BIO *bio_in) {
+  if (x509_) {
+    // TODO(AlCutter): Use custom deallocator
+    X509_free(x509_);
+    x509_ = nullptr;
+  }
+
+  x509_ = d2i_X509_bio(bio_in, &x509_);
+
+  if (!x509_) {
+    // At this point most likely the input was just corrupt. There are few
+    // real errors that may have happened (a malloc failure is one) and it is
+    // virtually impossible to fish them out.
+    LOG(WARNING) << "Input is not a valid encoded certificate";
+    LOG_OPENSSL_ERRORS(WARNING);
```

Comment:
This is part of the next change for CMS unwrapping, it's not used yet which is why this got through.